main.go

```diff
@@ -133,26 +43,49 @@ func main() {
 				log.Fatalf("couldn't close file %v, %v\n", *destFile, err)
 			}
 		}()
-
 		dst = dstFile
 	}
 
-	w := bufio.NewWriterSize(dst, *writeBufferSize)
+	data, err := mm.Map(src, mm.RDONLY, 0)
+	if err != nil {
+		log.Fatalf("couldn't mmap file: %v\n", err)
+	}
+	if len(data) <= 0 {
+		return
+	}
+
+	var str []string
 
+	// identify lines in data
+	p := 0
+	for i := 0; i < len(data); i++ {
+		if data[i] == '\n' {
+			str = append(str, sixb.BtS(data[p:i+1]))
```

Comment:
are you not worried about allocations here? I found that 

```
	for i := 0; i < len(m); i++ {	
		if m[i] == newLine {	
			numLines++	
		}	
	}	

	lines := Lines{data: m, slices: make([]Slice, 0, numLines)
```

speeds up data loading A LOT

Reply:
append is smart enough to amortize allocation cost. this way you only pass through data once.

Reply:
`str = append(str, sixb.BtS(data[p:i+1]))`

this grows the str array by x2 everytime we need more memory, making loading very slow and memory hungry. I will go back to preallocating.

Reply:
I can't load my data with this `append` approach - malloc complains. This why I ended up adding capacity.

Reply:
(on 102gb of ram, file size - 44gb..)

Reply:
hmm, maybe GOGC=30 will help

Reply:
I'll just leave with an extra pass through data for now instead of relying on GC config...

Reply:
It's loaded now, the overhead of string slices is same as I had before, needs 16 bytes per line (I presume 8 for len and 8 for capacity? not sure)

Reply:
string header occupies 16 bytes, yes same size with your initial Slice type

Reply:
thanks for confirming. How did you find this out? source code of go?

Reply:
unsafe.Sizeof("") is 16, you also find more info in reflect package.

Reply:
ah must be 8 bytes for start of the data array and 8 bytes for length